File: fizzbuzz-noiter.py

for i in range(1, 100):
  fizz = ""
  buzz = ""
  if i % 3 == 0:
    fizz = "fizz"
  if i % 5 == 0:
    buzz = "buzz"
  if fizz == "" and buzz == "":
    print i
  else:
    print "{}{}".format(fizz, buzz)

File: fizzbuzz-noiter.rs
/*
  Normal procedural fizzbuzz
*/

fn main() {
  for i in 1..100 {
    let mut fizz = "";
    let mut buzz = "";
    if i % 3 == 0 {
      fizz = "fizz";
    }
    if i % 5 == 0 {
      buzz = "buzz";
    }
    if fizz == "" && buzz == "" {
      println!("{}", i);
    } else {
      println!("{}{}", fizz, buzz);
    }
  }
}
